x-pack/plugins/apm/public/components/shared/charts/timeseries_chart.tsx

```diff
@@ -170,37 +172,26 @@ export function TimeseriesChart({
               data={isEmpty ? [] : serie.data}
               color={serie.color}
               curve={CurveType.CURVE_MONOTONE_X}
+              hideInLegend={serie.hideLegend}
+              fit={serie.fit ?? undefined}
+              filterSeriesInTooltip={
+                serie.hideTooltipValue ? () => false : undefined
+              }
```

Comment:
That's what I think we should do - to that end I've created APMChartSpec. I've kept TimeSeries as an alias to limit the amount of changes in this PR. It might still make sense to have some abstractions on top of Elastic Charts' `Spec` type, like this one. `filterSeriesInTooltip` might be too low level and perhaps there are others.